### DISCUSSION TITLE
Any chance of enabling marlin-binary-protocol

### DISCUSSION THREAD

**Discussion by JimKnightster** [Jyers/Marlin#171] [Ideas]
This would give us the ability to update any releases from Octoprint if/when you update the pre-compiled binaries again.  

Not a big deal but would be nice.

**Comment by Jyers** [comment]
I have been interested in updating the firmware through octoprint for a while, but I was totally unaware of this option. I'm definitely going to enable this in future precompiled versions!

**Comment by t3chguy** [comment]
One thing to keep in mind is that it is incompatible with Meatpack

**Reply by Jyers** [reply]
Meatpack seems to still be broken with the current marlin version, so this shouldn't be an issue for now.

**Reply by t3chguy** [reply]
I've been using it for the past two weeks with the only issue being sometimes the initial Connect fails and the printer needs to be restarted

![image](https://user-images.githubusercontent.com/2403652/110002634-812a1e00-7d0d-11eb-81c5-7e0e3a4f819f.png)

**Reply by Jyers** [reply]
Interesting, I could not for the life of me even get it to connect. If I may, what exactly is the result of the incompatibility here? Does meatpack stop working with this setting enabled or will binary transfer not work with meatpack enabled?

**Reply by t3chguy** [reply]
Compilation failure

**Reply by Jyers** [reply]
Wait... Is there a meatpack config option I am unaware of? That would explain why I can't get it to work.

**Reply by t3chguy** [reply]
Yeah, you need a `#define MEATPACK` in your Marlin config
https://github.com/MarlinFirmware/Marlin/blob/bugfix-2.0.x/Marlin/Configuration_adv.h#L3332

**Reply by Jyers** [reply]
Hmmm, that would explain it. Its too bad they conflict, but considering meatpack isn't enabled in the precompiled versions, I see no reason not to enable binary file transfer
